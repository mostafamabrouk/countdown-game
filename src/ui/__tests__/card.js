/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import { Card } from "../";
import "jest-styled-components";

test("Card renders withouut any parameters", () => {
  const component = renderer.create(<Card />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Card renders with character", () => {
  const component = renderer.create(<Card char="s" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Card renders with character and error", () => {
  const component = renderer.create(<Card char="s" error={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Card renders with character and success", () => {
  const component = renderer.create(<Card char="s" success={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Card renders with character and dragged before", () => {
  const component = renderer.create(<Card char="s" opaque={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
