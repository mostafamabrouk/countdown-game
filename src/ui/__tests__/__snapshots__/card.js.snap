// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders with character 1`] = `
.c0 {
  width: 120px;
  height: 160px;
  border: 2px solid #000;
  margin: 0 8px;
  text-transform: uppercase;
  line-height: 160px;
  font-family: Helvetica;
  font-size: 96px;
  color: #009680;
  text-align: center;
  cursor: move;
  -webkit-transition: 0.2s all ease;
  transition: 0.2s all ease;
}

<div
  className="c0"
>
  s
</div>
`;

exports[`Card renders with character and dragged before 1`] = `
.c0 {
  width: 120px;
  height: 160px;
  border: 2px solid #000;
  margin: 0 8px;
  text-transform: uppercase;
  line-height: 160px;
  font-family: Helvetica;
  font-size: 96px;
  color: #009680;
  text-align: center;
  cursor: move;
  -webkit-transition: 0.2s all ease;
  transition: 0.2s all ease;
  opacity: 0.25;
}

<div
  className="c0"
>
  s
</div>
`;

exports[`Card renders with character and error 1`] = `
.c0 {
  width: 120px;
  height: 160px;
  border: 2px solid #000;
  margin: 0 8px;
  text-transform: uppercase;
  line-height: 160px;
  font-family: Helvetica;
  font-size: 96px;
  color: #009680;
  text-align: center;
  cursor: move;
  -webkit-transition: 0.2s all ease;
  transition: 0.2s all ease;
  border-color: #FF0000;
  opacity: 0.42;
}

<div
  className="c0"
>
  s
</div>
`;

exports[`Card renders with character and success 1`] = `
.c0 {
  width: 120px;
  height: 160px;
  border: 2px solid #000;
  margin: 0 8px;
  text-transform: uppercase;
  line-height: 160px;
  font-family: Helvetica;
  font-size: 96px;
  color: #009680;
  text-align: center;
  cursor: move;
  -webkit-transition: 0.2s all ease;
  transition: 0.2s all ease;
  border-color: #6CBC36;
  color: #000;
}

<div
  className="c0"
>
  s
</div>
`;

exports[`Card renders withouut any parameters 1`] = `
.c0 {
  width: 120px;
  height: 160px;
  border: 2px solid #000;
  margin: 0 8px;
  text-transform: uppercase;
  line-height: 160px;
  font-family: Helvetica;
  font-size: 96px;
  color: #009680;
  text-align: center;
  cursor: move;
  -webkit-transition: 0.2s all ease;
  transition: 0.2s all ease;
}

<div
  className="c0"
/>
`;
